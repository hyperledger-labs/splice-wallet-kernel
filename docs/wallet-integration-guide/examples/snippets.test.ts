import { test, expect, jest } from '@jest/globals'
import { readdirSync, readFileSync } from 'fs'
import path from 'path'

// WARNING: the `allocate-party.ts` snippet does not succeed if ran 2 or more times without stopping and restarting localnet (as the same mocked party identity is then already onboarded)

const mockKeyPair = {
    publicKey: '+5E6ccwzeYq0CEyjSoIQNNFq8v31xCfVCm+PxSXyUxs=',
    privateKey:
        'oQ+sEiZeiBtPrhneCtlceaW6th6M6jyg1xQroaTiQP77kTpxzDN5irQITKNKghA00Wry/fXEJ9UKb4/FJfJTGw==',
}

jest.mock('uuid', () => {
    return {
        v4: jest.fn().mockReturnValue('6e747c98-2d9a-4ec6-8e22-c4bc40948c41'),
    }
})

jest.mock('@canton-network/core-signing-lib', () => {
    // Require the original module to not be mocked...
    const originalModule = jest.requireActual(
        '@canton-network/core-signing-lib'
    ) as any

    return {
        __esModule: true, // Use it when dealing with esModules
        ...originalModule,
        createKeyPair: jest.fn().mockReturnValue(mockKeyPair),
    }
})

describe('testing doc snippets', () => {
    const snippets = readdirSync('./snippets').filter((f) => f.endsWith('.ts'))

    for (const filename of snippets) {
        const fullpath = path.join(__dirname, './snippets', filename)
        const contents = readFileSync(fullpath)

        if (contents.includes('// @disable-snapshot-test')) {
            continue
        }

        test(
            filename,
            async () => {
                const mod = await import(fullpath.replace('.ts', '.js'))
                expect(mod.default).toBeDefined()

                const result = await mod.default()

                // Run `yarn jest -u` to update snapshots for new changes
                expect(result).toMatchSnapshot()
            },
            10000
        )
    }
})
