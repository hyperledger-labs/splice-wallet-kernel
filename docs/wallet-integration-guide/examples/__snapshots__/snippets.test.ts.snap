// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing doc snippets allocate-party.ts 1`] = `"my-wallet-1::122078180903ec6fe58fe30bec7380b351fde137d86eeaea837f6d379a86a8039885"`;

exports[`testing doc snippets create-key-pair.ts 1`] = `
{
  "privateKey": "oQ+sEiZeiBtPrhneCtlceaW6th6M6jyg1xQroaTiQP77kTpxzDN5irQITKNKghA00Wry/fXEJ9UKb4/FJfJTGw==",
  "publicKey": "+5E6ccwzeYq0CEyjSoIQNNFq8v31xCfVCm+PxSXyUxs=",
}
`;

exports[`testing doc snippets generate-fingerprint.ts 1`] = `"1220dd3caa6464d2ba93b1fb195ede9344d54a0e17405d9331e91ceea27f07d414f7"`;
